### PR TITLE
client/core: Refactor rotateBonds

### PR DESCRIPTION
**Pull request by martonp**
This changes no functionality, just extracts the login in rotateBonds into multiple smaller functions.

It was a bit daunting to add the logic for updating existing ETH bonds. This will make it simpler.

### CODE REVIEW

File: client/core/bond.go [hunk 1]
```diff
@@ -150,337 +150,304 @@ func minBondLifetime(net dex.Network, bondExpiry int64) time.Duration {
...
+		assetIDs[assetID] = struct{}{}
+	}
+
+	return assetIDs, spentBonds, nil
+}
+
+// clearSpentBonds removes all bonds that have been successfully refunded from
+// a dexConnection's expiredBonds list and returns the strength of the remaining
+// bonds.
+func (c *Core) clearSpentBonds(dc *dexConnection, bondCfg *dexBondCfg, spentBonds []*bondID) int64 {
```
Comment (chappjc) [inline client/core/bond.go:395]: Why does this one need to be separate, rather than part of `refundExpiredBonds`?

File: client/core/bond.go [hunk 2]
```diff
@@ -150,337 +150,304 @@ func minBondLifetime(net dex.Network, bondExpiry int64) time.Duration {
...
+		// tierChange field, but it could change, so we remember it.
+		actuatedTierChanges[state.bondAssetID] = append(actuatedTierChanges[state.bondAssetID],
+			acctUpdate{dc.acct, state.tierChange, reserveDelta},
+		)
+	}
+}
 
+// applyReserveAdjustments notifies wallets of the aggregate reserve
+// adjustments over all dex connections.
+func (c *Core) applyReserveAdjustments(netReserveDeltas map[uint32]int64, actuatedTierChanges map[uint32][]acctUpdate) {
```
Comment (chappjc) [inline client/core/bond.go:450]: This body of this reserves helper and `recordReserveAdjustments` look better suited to stay in `rotateBonds`.  Is this also important for ETH bond renewal/extension?

**Review by chappjc — APPROVED** [review]
I honestly miss the linearly flowing code, but this looks fine and I think I understand the need.
There don't appear to be any issues or change of behavior with this.

File: client/core/bond.go (hunk 1 above)
Comment (martonp) [reply client/core/bond.go]: No need, I've moved it.

File: client/core/bond.go (hunk 2 above)
Comment (martonp) [reply client/core/bond.go:444]: I've moved it back there. The point if this was that this function was already ~500 lines and the update bond code will also add a good amount of additional logic. Personally I think it's very hard to wrap my head around these huge functions. If the details are extracted into smaller helper functions, it's easier to quickly know what is happening and make changes. Why do you like the linear flowing code better?

Comment (chappjc) [reply client/core/bond.go:444]: > Why do you like the linear flowing code better?

Just readable for me is all.

Regarding the reserves adjustments, when I see variables like `netReserveDeltas` and `actuatedTierChanges` (and these two methods that need them) that are only instantiated or used in one other method, it stands out.  The functionality in either of these methods is useless unless applied exactly how `rotateBonds` applies it, and they'd always be used in succession.  So for me it's confusing (bounce around the code and wonder why they are separate methods), and it adds more the `Core`'s method set, which is not such a good thing even if they are not exported.

File: client/core/bond.go [hunk 3]
```diff
@@ -580,11 +573,38 @@ func (c *Core) rotateBonds(ctx context.Context) {
...
+			if err != nil { // we grabbed it above so shouldn't happen
+				c.log.Errorf("%v wallet not available for bond: %v", unbip(assetID), err)
+				continue
+			}
+			c.log.Infof("Updating bond reserves by %s (automatic tier change adjustments)",
+				wallet.amtStringSigned(reserveDelta))
+			wallet.ReserveBondFunds(reserveDelta, false)
+		}
+		// Update the unactuated tier change counter.
+		for _, s := range actuatedTierChanges[assetID] {
```
Comment (chappjc) [inline client/core/bond.go]: Thanks for switching this back.  The `netReserveDeltas` and `actuatedTierChanges` variables are just naturally locally scoped, and the helper methods shouldn't be used in any other context, so they are baggage and make it hard to see what's really being done.  I think these lines are not so long that it makes `rotateBonds` unreadable for you.  I respect the differences of opinion, and I do think the other refactoring mostly helps.

Comment (chappjc) [reply client/core/bond.go]: Oh, looks like you need to remove the two helper functions.

**Review by chappjc — APPROVED** [review]

**Pull request merged**